src/vaadin-combo-box-mixin.html

```diff
@@ -352,8 +352,9 @@
     _onClick(e) {
       this._closeOnBlurIsPrevented = true;
 
+      const clearButton = this.$.input ? this.$.input.$.clearButton : this._clearElement;
       const path = e.composedPath();
-      if (path.indexOf(this._clearElement) !== -1) {
+      if (path.indexOf(clearButton) !== -1) {
         this._clear();
```

Comment:
BTW Clearing is already implemented in the text-field itself. That suggests that, ideally, we should remove the clear implementation from here altogether, I think.

I do understand that it might be not possible to do right now, so it’s fine to clean that up later. Let me kindly ask for a TODO comment here to remind us about that though.

Reply:
Yeah I think it would be good if it worked in a way that text field just clears its value which gets passed on, but I think it didn't work so simply right now so that this was needed to make it work for now. I can add a comment about it tomorrow.

Reply:
I tried testing if it would be possible to get this working without having at least some clear logic in either the mixin or both vaadin-combo-box and vaadin-combo-box-light, but it seems that's not really possible. We need at least the focus call after clearing, even if it would be possible to get the value to update automatically.

I tried separating the custom clear button behaviour to vaadin-combo-box-light.html and having the vaadin-text-field clear button related logic in vaadin-combo-box.html, but I wasn't able to make it any cleaner than it was so I left this bit as is in the mixin code. I didn't add a TODO comment since I'm not sure what could be done about it.